confluent_kafka/avro/serializer/message_serializer.py

```diff
@@ -143,7 +143,7 @@ def encode_record_with_schema_id(self, schema_id, record, is_key=False):
         writer = self.id_to_writers[schema_id]
         with ContextStringIO() as outf:
             # Write the magic byte and schema ID in network byte order (big endian)
-            outf.write(struct.pack('>bI', MAGIC_BYTE, schema_id))
+            outf.write(struct.pack('>bi', MAGIC_BYTE, schema_id))
```

Comment:
Should we even attempt to make a request with a negative id? Sounds like some previous error handling was insufficient if we get here with an invalid schema.

Reply:
No we should not the negative value is internal to SR and should never be used by clients. That said in order to properly conform with the spec we should be using a signed int.